Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -174,9 +182,69 @@ namespace OpenInfraPlatform
 
 					BLUE_LOG(warning) << "Geometry conversion for IfcSectionedSolidHorizontal not implemented.";
 					// TO DO: implement, check for formal propositions. 
+                 
 
-				 //endif sectioned_solid_horizontal
-			}
+				//check dimensions and correct attributes sizes
+                if(vec_cross_sections.size() != vec_cross_section_positions.size())
+				{
+					std::cout << "CrossSections and CrossSectionsPositions are not equal in size " << std::endl;
+					return;
+				}
+				if ((vec_cross_sections.size() - vec_cross_section_positions.size()) > 1)
+				{
+					std::cout << "vec_cross_sections size decreased by one" << std::endl;
+					vec_cross_sections.pop_back();
+				}
+				else if ((vec_cross_section_positions.size() - vec_cross_sections.size()) > 1)
+				{
+					std::cout << "vec_cross_section_positions decreased by one" << std::endl;
+					vec_cross_section_positions.pop_back();
+				}
+
+                 //TO DO: check lenght conversions on the ProfileDistanceExpression L[2:?] CrossSectionPostitions
+				double length_in_meter = UnitConvert()->getLengthInMeterFactor();
+
+
+				 //Give directrix to Curve converter
+		    	 std::vector<carve::geom::vector<3> > segment_start_points;
+	     		 std::vector<carve::geom::vector<3> > basis_curve_points;
+				 curveConverter->convertIfcCurve(directrix.lock(), basis_curve_points, segment_start_points);
+
+				 std::shared_ptr<carve::input::PolyhedronData> pipe_data(new carve::input::PolyhedronData());

```

Comment:
It's not really a pipe, right? Perhaps a better name would be `body_data` or sth like that?

Reply:
Please mark the conversation "resolved" when finished - it enables an oversight, what is finished and what not.

Reply:
resolved in commit ccd8948